orc/src/main/java/org/apache/iceberg/orc/ORCSchemaUtil.java

```diff
@@ -296,24 +304,36 @@ private static TypeDescription buildOrcProjection(
                 list.elementId(),
                 list.elementType(),
                 isRequired && list.isElementRequired(),
-                mapping);
+                mapping,
+                config);
         orcType = TypeDescription.createList(elementType);
         break;
       case MAP:
         Types.MapType map = (Types.MapType) type;
         TypeDescription keyType =
-            buildOrcProjection(map.keyId(), map.keyType(), isRequired, mapping);
+            buildOrcProjection(map.keyId(), map.keyType(), isRequired, mapping, config);
         TypeDescription valueType =
             buildOrcProjection(
-                map.valueId(), map.valueType(), isRequired && map.isValueRequired(), mapping);
+                map.valueId(),
+                map.valueType(),
+                isRequired && map.isValueRequired(),
+                mapping,
+                config);
         orcType = TypeDescription.createMap(keyType, valueType);
         break;
       default:
         if (mapping.containsKey(fieldId)) {
           TypeDescription originalType = mapping.get(fieldId).type();
           Optional<TypeDescription> promotedType = getPromotedType(type, originalType);
 
-          if (promotedType.isPresent()) {
+          boolean convertTimestampTZ =
+              config.getBoolean(ConfigProperties.ORC_CONVERT_TIMESTAMPTZ, false);
+
+          if (convertTimestampTZ
+              && type.typeId() == Type.TypeID.TIMESTAMP
```

Comment:
Guess this deserves a new `case TIMESTAMP`, not an `if` here.

Reply:
Why is this logic is not part of the `getPromotedType` method?